de.fu_berlin.inf.dpp.intellij/src/de/fu_berlin/inf/dpp/intellij/editor/annotations/AnnotationQueue.java

```diff
@@ -0,0 +1,138 @@
+package de.fu_berlin.inf.dpp.intellij.editor.annotations;
+
+import de.fu_berlin.inf.dpp.filesystem.IFile;
+import de.fu_berlin.inf.dpp.session.User;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.ArrayDeque;
+import java.util.List;
+import java.util.Queue;
+
+/**
+ * A class to store a limited number of annotations of a certain type. The store
+ * operates like a queue, meaning, if the store is full, the oldest annotation
+ * is removed. This is not done automatically but rather should be done by the
+ * caller by calling {@link #removeIfFull()} before adding new annotations.
+ *
+ * @param <E> the stored annotation type
+ */
+class AnnotationQueue<E extends AbstractEditorAnnotation>
+    extends AnnotationStore<E> {
+
+    private final Queue<E> annotationQueue;
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Rename field "annotationQueue" ([squid:S1700](https://sonarcloud.io/organizations/saros/rules#rule_key=squid%3AS1700))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=saros_complete&pullRequest=238&issues=AWYa7GVavOWp7Yd62c1R&open=AWYa7GVavOWp7Yd62c1R)

[ISSUE_KEY]: AWYa7GVavOWp7Yd62c1R